storagenode/piecestore/endpoint.go

```diff
@@ -315,6 +318,14 @@ func (endpoint *Endpoint) Upload(stream pb.DRPCPiecestore_UploadStream) (err err
 	defer commitOrderToStore(ctx, &largestOrder)
 
 	for {
+		// Measure upload rate (bytes per second)
+		dt := time.Since(startTime)
```

Comment:
note, you can move this into isSlowUpload check as well

Reply:
wip